pyproject.toml

```diff
@@ -1,12 +1,20 @@
 [build-system]
-requires = ["poetry"]
+requires = [
+    "poetry",
+    # Starting with NumPy 1.25, NumPy is (by default) as far back compatible
+    # as oldest-support-numpy was (customizable with a NPY_TARGET_VERSION
+    # define).  For older Python versions (where NumPy 1.25 is not yet avaiable)
+    # continue using oldest-support-numpy.
+    "oldest-supported-numpy>=0.14; python_version<'3.9'",
+    "numpy>=1.25; python_version>='3.9'",
```

Comment:
This is necessary due to the addition of pyarrow, which depends on numpy. Once we stop supporting Python 3.8, this can be simplified